packages/edit-widgets/src/components/layout/interface.js

```diff
@@ -82,6 +99,21 @@ function Interface( { blockEditorSettings } ) {
 		}
 	}, [ isInserterOpened, isListViewOpened, isHugeViewport ] );
 
+	useEffect( () => {
+		if ( isInserterOpened && ! isLastSelectedWidgetAreaOpen ) {
+			// Select the last selected block if hasn't already.
+			selectBlock( widgetAreaClientId );
+			// Open the last selected widget area when opening the inserter.
+			setIsWidgetAreaOpen( widgetAreaClientId, true );
+		}
+	}, [
+		widgetAreaClientId,
+		isInserterOpened,
+		isLastSelectedWidgetAreaOpen,
+		selectBlock,
+		setIsWidgetAreaOpen,
+	] );
```

Comment:
The unique processing that existed in the `DocumentTools` component of the Widget Editor has been moved here. If the selected widget area is closed when you click the inserter button, the widget area will be opened.

![a5a85f0d3fc3ea443426f1891966f99d](https://github.com/WordPress/gutenberg/assets/54422211/f8810d9f-d053-49bd-9f49-d843042243b9)